misc/review/MidiEndpointDeviceInformation.idl

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+// ============================================================================
+// This is part of the Windows MIDI Services App API and should be used
+// in your Windows application via an official binary distribution.
+// Further information: https://github.com/microsoft/MIDI/
+// ============================================================================
+
+#include "namespace_defines.h"
+#include "midl_defines.h"
+MIDI_IDL_IMPORT
+
+import "MidiGroupTerminalBlock.idl";
+import "MidiProtocolEnum.idl";
+import "MidiEndpointNativeDataFormatEnum.idl";
+import "MidiEndpointDevicePurposeEnum.idl";
+import "MidiFunctionBlock.idl";
+
+import "MidiEndpointDeviceInformationSortOrderEnum.idl";
+import "MidiEndpointDeviceInformationFiltersEnum.idl";
+
+namespace MIDI_MIDL_NAMESPACE
+{
+    MIDI_API_CONTRACT_v1
+    [default_interface]
+    runtimeclass MidiEndpointDeviceInformation
+    {
+        static MidiEndpointDeviceInformation CreateFromId(String id);
+
+        static IVectorView<MidiEndpointDeviceInformation> FindAll();
+        static IVectorView<MidiEndpointDeviceInformation> FindAll(MidiEndpointDeviceInformationSortOrder sortOrder);
+        static IVectorView<MidiEndpointDeviceInformation> FindAll(MidiEndpointDeviceInformationSortOrder sortOrder, MidiEndpointDeviceInformationFilters endpointFilters);
+
+        static String DiagnosticsLoopbackAEndpointId{ get; };
+        static String DiagnosticsLoopbackBEndpointId{ get; };
+
+        static String EndpointInterfaceClass{ get; };
```

Comment:
Same way device watcher formats it. It's a stringified GUID, because DeviceInformation.Id() is always an hstring, but then the actual property on the MidiEndpointDeviceInformation object is GUID strong type.